misc/python/materialize/checks/checks.py

```diff
@@ -19,9 +19,6 @@ class Check:
     def __init__(self, base_version: MzVersion, rng: Optional[Random]) -> None:
         self.base_version = base_version
         self.rng = rng
-        self._initialize = self.initialize()
```

Comment:
This is a rather large change in the design of platform-checks. By calling initialize/manipulate/validate only at the time we want to execute the code, we can finally use the actual version of the database. This should make tests much more straightforward to write while a feature is in active development.